src/Cache.ts

```diff
@@ -49,7 +49,9 @@ export class Cache {
       d('* Replacing existing file');
       await fs.promises.rm(cachePath, { recursive: true, force: true });
     }
-    await fs.promises.rename(currentPath, cachePath);
+
+    await fs.promises.copyFile(currentPath, cachePath);
+    await fs.promises.rm(currentPath);
```

Comment:
The performance of rename is better than cp + rm. We prioritize using rename, and only when rename fails with an error of EXDEV do we fallback to cp + rm.

```suggestion
    try {
      await fs.promises.rename(currentPath, cachePath);
    } catch (err) {
      if ((err as NodeJS.ErrnoException).code !== "EXDEV") {
        throw err;
      }
      
      await fs.promises.cp(currentPath, cachePath, { recursive: true, force: true });
      await fs.promises.rm(currentPath, { recursive: true, force: true });
    }
```